plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -682,14 +682,15 @@ enterComputeCek = computeCek (toWordArray 0) where
         case sch of
             -- It's only possible to apply a builtin application if the builtin expects a term
             -- argument next.
-            TypeSchemeArrow _ schB -> do
-                x <- readKnown (Just argTerm) arg
-                -- TODO: should we bother computing that 'ExMemory' eagerly? We may not need it.
-                -- We pattern match on @arg@ twice: in 'readKnown' and in 'toExMemory'.
-                -- Maybe we could fuse the two?
-                let runtime' = BuiltinRuntime schB (f x) . exF $ toExMemory arg
-                res <- evalBuiltinApp fun term' env runtime'
-                returnCek unbudgetedSteps ctx res
+            TypeSchemeArrow _ schB -> case readKnown (Just argTerm) arg of
```

Comment:
I'd be curious to see what happens if you do it this way:
```
do
  x <- liftEither $ readKnown ...
  ...
```
That might force us back into ST (if it doesn't get optimized away, which it probably will), which might be enlightening. And if it does get optimized away, it's arguably better style.

Reply:
Great suggestion, thank you.